### PR TITLE
ICU-21249 ICU4C API promotions

### PR DESCRIPTION
**Pull request by richgillam**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-21249
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [ ] Tests included
- [x] Documentation is changed or added

Updated the `@draft ICU 65` documentation tags to `@stable ICU 65`.  See https://docs.google.com/spreadsheets/d/1JNV3j4qu_ezpKqXRWTFfyA-Jscss9tqowd-tLTl8dWM/edit#gid=0 for details.

### CODE REVIEW

**Comment by markusicu** [comment]
The Travis CI failure shows that there is an inconsistency between @draft/@stable tags and their corresponding ifndef HIDE... guards. Please fix. https://travis-ci.org/github/unicode-org/icu/jobs/729813997

File: icu4c/source/common/unicode/localematcher.h [hunk 1]
```diff
@@ -461,7 +460,7 @@ class U_COMMON_API LocaleMatcher : public UMemory {
          *
          * @param demotion the demotion per desired locale to set.
          * @return this Builder object
-         * @draft ICU 65
+         * @stable ICU 65
          */
         Builder &setDemotionPerDesiredLocale(ULocMatchDemotion demotion);
```
Comment (markusicu) [inline icu4c/source/common/unicode/localematcher.h:466]: Please add a new ifndef U_HIDE_DRAFT_API guard around the following setDirection() which is draft ICU 67.

Comment (richgillam) [reply icu4c/source/common/unicode/localematcher.h:466]: Sorry about that.  Hopefully fixed now.

**Review by markusicu — COMMENTED** [review]
lgtm please squash

I will also ask others if they want to review.

**Comment by markusicu** [comment]
"This branch has conflicts that must be resolved" -- please git rebase master & merge & force-push...

File: icu4c/source/common/unicode/localematcher.h [hunk 2]
```diff
@@ -20,26 +20,24 @@
...
  */
 
-#ifndef U_FORCE_HIDE_DRAFT_API
-
 /**
  * Builder option for whether the language subtag or the script subtag is most important.
  *
- * @see LocaleMatcher::Builder#setFavorSubtag(ULocMatchFavorSubtag)
- * @draft ICU 65
+ * @see Builder#setFavorSubtag(ULocMatchFavorSubtag)
```
Comment (markusicu) [inline icu4c/source/common/unicode/localematcher.h:26]: Looks like you didn't merge, you are now just clobbering my earlier changes. Please fix.

Comment (richgillam) [reply icu4c/source/common/unicode/localematcher.h:26]: Sorry; I should have paid better attention.  Let me try to figure out how to get those changes back now...

Comment (markusicu) [reply icu4c/source/common/unicode/localematcher.h:26]: For reference: PR #1368

Comment (richgillam) [reply icu4c/source/common/unicode/localematcher.h]: Thanks for that; hopefully fixed now.  Apologies once again for not looking more closely.

File: icu4c/source/common/unicode/localematcher.h [hunk 3]
```diff
@@ -298,15 +298,16 @@ class U_COMMON_API LocaleMatcher : public UMemory {
      * LocaleMatcher builder.
      * Movable but not copyable.
      *
-     * @draft ICU 65
+     * @see LocaleMatcher#builder()
```
Comment (markusicu) [inline icu4c/source/common/unicode/localematcher.h:301]: remove this line too (mis-merged)

Comment (richgillam) [reply icu4c/source/common/unicode/localematcher.h]: Done.  Apologies once again...

**Review by markusicu — APPROVED** [review]
lgtm tnx

Please wait for the build bots to settle down (except maybe AppVeyor which can take a long time and is flaky).

**Pull request merged**